opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -161,56 +175,40 @@ def force_flush(self) -> bool:
 
         return metric_reader_result and metric_exporter_result
 
-    def register_metric_reader(self, metric_reader: "MetricReader") -> None:
-        # FIXME protect this method against race conditions
-        self._metric_readers.append(metric_reader)
-
-    def register_metric_exporter(
-        self, metric_exporter: "MetricExporter"
-    ) -> None:
-        # FIXME protect this method against race conditions
-        self._metric_exporters.append(metric_exporter)
-
-    def register_view(self, view: "View") -> None:
-        # FIXME protect this method against race conditions
-        self._views.append(view)
-
-
-class MetricReader(ABC):
-    def __init__(self):
-        self._shutdown = False
-
-    @abstractmethod
-    def collect(self):
-        pass
-
     def shutdown(self):
-        # FIXME this will need a Once wrapper
-        self._shutdown = True
+        # FIXME implement a timeout
 
+        if self._shutdown:
+            _logger.warning("shutdown can only be called once")
+            return False
 
-class MetricExporter(ABC):
-    def __init__(self):
-        self._shutdown = False
+        result = True
 
-    @abstractmethod
-    def export(self):
-        pass
+        for metric_reader in self._metric_readers:
+            result = result and metric_reader.shutdown()
```

Comment:
Should we have a warning is reader result returns false?

Reply:
Sure, added a warning.